packages/ethash/src/index.ts

```diff
@@ -11,9 +11,127 @@ import {
 } from './util'
 // eslint-disable-next-line implicit-dependencies/no-implicit
 import type { LevelUp } from 'levelup'
-import type { Block, BlockHeader } from '@ethereumjs/block'
+import { Block, BlockHeader } from '@ethereumjs/block'
 const xor = require('buffer-xor')
 
+type Solution = {
+  mixHash: Buffer
+  nonce: Buffer
+}
+
+class Miner {
+  public blockHeader: BlockHeader
+  public block?: Block
+  public ethash: Ethash
+
+  public currentNonce: BN
+  public solution?: Solution
+
+  private headerHash?: Buffer
+  private stopMining: boolean
+
+  /**
+   * Create a Miner object
+   * @param mineObject - The object to mine on, either a `BlockHeader` or a `Block` object
+   * @param ethash - Ethash object to use for mining
+   */
+
+  constructor(mineObject: BlockHeader | Block, ethash: Ethash) {
+    if (mineObject.constructor === BlockHeader) {
```

Comment:
just curious, is there a reason you chose to compare it this way over using `instanceof`?